runtime/common/orderedmap/orderedmap.go

```diff
@@ -45,6 +45,15 @@ func NewKeyTypeValueTypeOrderedMap() *KeyTypeValueTypeOrderedMap {
 	}
 }
 
+// Clear removes all entries from this ordered map.
+func (om *KeyTypeValueTypeOrderedMap) Clear() {
+	om.list.Init()
+	// NOTE: Range over map is safe, as it is only used to delete entries
+	for key := range om.pairs { //nolint:maprangecheck
+		delete(om.pairs, key)
+	}
```

Comment:
would it be faster if we re-initialized the map, and let the old one be GC'ed?
```suggestion
	om.pairs = make(map[KeyType]*KeyTypeValueTypePair)
```

Reply:
The idea here is to reuse the map and avoid allocating more memory and producing garbage.
I benchmarked this suggestion and it's both less memory efficient, but also slower:

```
clearing by deleting:	7682	    147602 ns/op	   68704 B/op	    1193 allocs/op
clearing by new map:	7294	    153780 ns/op	   78146 B/op	    1283 allocs/op
```

Good idea though, I was curious here how Go performs

Reply:
oh nice, interesting results!